core/src/main/java/hudson/triggers/Trigger.java

```diff
@@ -156,7 +158,7 @@ public TriggerDescriptor getDescriptor() {
      * periodically. This is useful when your trigger does
      * some polling work.
      */
-    protected Trigger(String cronTabSpec) throws ANTLRException {
+    protected Trigger(@Nonnull String cronTabSpec) throws ANTLRException {
```

Comment:
parameters should be "not nullable" by default unless a "this can be null" annotation is provided.